corgi/collectors/prod_defs.py

```diff
@@ -18,17 +18,40 @@ def get_product_definitions(cls) -> dict:
         return response.json()
 
     @classmethod
-    def load_product_definitions(cls) -> list:
-        data = cls.get_product_definitions()
+    def get_community_product_definitions(cls) -> dict:
+        with open("config/community_product_definitions.json") as proddefs_data:
+            return json.load(proddefs_data)
 
-        products = []
+    @classmethod
+    def load_products(cls, data: dict) -> list[dict]:
+        products: list[dict] = []
         for ps_product, product in data["ps_products"].items():
-            if product["business_unit"] == "Community" and not settings.COMMUNITY_PRODUCTS_ENABLED:
-                continue
+            if settings.COMMUNITY_MODE_ENABLED:
+                if product["business_unit"] != "Community":
+                    continue
+            else:  # Enterprise mode
+                if product["business_unit"] == "Community":
+                    continue
             product["id"] = ps_product
             products.append(product)
 
+        return products
+
+    @classmethod
+    def load_product_definitions(cls) -> list:
+        try:
+            data = cls.get_product_definitions_service()
+        except ConnectionError as e:
+            if settings.COMMUNITY_MODE_ENABLED:
+                data = cls.get_community_product_definitions()
+            else:
+                raise e
```

Comment:
This looks a little odd. Maybe it's better to check the setting before we make a request:
```suggestion
        if settings.COMMUNITY_MODE_ENABLED:
            data = cls.get_community_product_definitions()
        else:  # Enterprise mode
            data = cls.get_product_definitions_service()
```

Reply:
I designed it that way originally. However that will put a burden on us to keep the local product definitions up to date. I would rather have it this way so that if we can access the internal dashboard we prefer to use it instead of the local copy. The local copy does allow the codebase to standalone outside of the RH firewall, but I think it's better to only use it where dashboard is not available.

Reply:
Thanks for explaining! Your approach here sounds good in that case.

EDIT: Actually, I just realized - this code is running in stage / prod as well as locally, right? So if there's some kind of outage or network glitch, stage might fall back to using the community product definitions with the current code. Maybe that's not an issue - if so, please feel free to mark this resolved.

Reply:
Good catch. I think that's still OK though as the code will only load products which don't have business unit equal to "Community", so no chance of loading community products into enterprise version. The prod_defs collector will simply return 0 product definitions, which should be a no-op I think.

Reply:
Also before loading local product definitions, we first check if COMMUNITY_MODE is ENABLED. If it's not enabled we just reraise the ConnectionError.